fix(throttleTime): double emission with leading and trailing enabled

MatthiasKunnen opened the pull request:
This PR fixes an issue with throttleTime emitting both leading and trailing values in the same time window.
    
Double emission problem:
```
source:   a123b12-c-23d-2-ef---
expected: a---b---c---d---e---f
actual:   a---b1---c2---2-e---f
```
    
Closes #2466 and closes #2727.
Follows #2749 and #2864.

benlesh commented:
I'm not sure this is the correct behavior. Even if it was, this would be a breaking change for someone.

I'll wait for other core team members to weigh in... @JWO719? @cartant? @kwonoj?

MatthiasKunnen commented:
I think you've already had a discussion about whether the behavior is intended in #2749. See https://github.com/ReactiveX/rxjs/pull/2749#discussion_r141075604.

That PR was closed by the author. Perhaps they didn't have the time.

MatthiasKunnen commented:
I've fixed the fact that slow user code could influence timings.

I understand that time is finite and this PR might not be anywhere on the priority list but would a review be possible?

I've also rebased the PR to be up-to-date.

adolgoff commented:
PR is definitely fixing unexpected behavior
